src/server/src/main/java/io/cassandrareaper/resources/ClusterResource.java

```diff
@@ -347,6 +347,23 @@ public Response deleteCluster(@PathParam("cluster_name") String clusterName)
           .entity("cluster \"" + clusterName + "\" not found")
           .build();
     }
+    if (force.orElse(Boolean.FALSE)) {
+      int attempts = 0;
+      do {
+        context.storage.getRepairSchedulesForCluster(clusterName)
+            .forEach((schedule) -> context.storage.deleteRepairSchedule(schedule.getId()));
+
+        context.storage.getRepairRunIdsForCluster(clusterName)
+            .forEach((uuid) -> context.storage.deleteRepairRun(uuid));
+
+        try {
+          Thread.sleep(100);
+        } catch (InterruptedException ignore) {
+          break;
+        }
+      } while (9 > attempts && (context.storage.getRepairSchedulesForCluster(clusterName).isEmpty()
+            || !context.storage.getRepairRunsForCluster(clusterName, Optional.empty()).isEmpty()));
+    }
```

Comment:
That condition is hard to read and understand.
You're looping as long as you haven't reached 10 attempts and [there are no more schedules or there are still repair runs].
Shouldn't the condition be [there are still schedules or there are still repair runs] ?